src/hypothesis/strategies.py

```diff
@@ -1155,23 +1155,36 @@ def decimals(min_value=None, max_value=None,
     # Set up a strategy for finite decimals
     if places is not None:
         # Fixed-point decimals are basically integers with a scale factor
+        # and annoying tendencies to break in unusual precision contexts.
+        def ctx(val):
+            """Return a context in which this value is lossless."""
+            precision = ceil(math.log10(abs(val or 1))) + places + 1
```

Comment:
I had to do a fair bit of staring at this and convincing myself it was correct, and it's only so because the math module is doing something interesting (I'm not sure what precisely) - if log10 were going via float then this would be an error for very large numbers, but it seems that there's a direct calculation for ints.

This might be too specific to even warrant a comment in the code, I don't know, I just thought I'd register that I was confused but believe this to be correct.